other-charts/prepull-daemonset/README.md.gotmpl

```diff
@@ -0,0 +1,36 @@
+{{ template "rstudio.header" . }}
+
+{{ template "chart.versionBadge" . }}{{ template "chart.typeBadge" . }}{{ template "chart.appVersionBadge" . }}
+
+{{ template "rstudio.description" . }}
+
+{{ template "rstudio.install" . }}
+
+## Use
+
+Create a `values.yaml` file somewhere that lists the images you'd like to be pre-pulled (use the `values.yaml` file in this repo for format):
+
+```
+images:
+  - name: my-first-image
+    image: some-id.dkr.ecr.us-east-1.amazonaws.com/my-ecr-repo:latest
+  - name: my-second-image
+    image: some-id.dkr.ecr.us-east-1.amazonaws.com/my-other-repo:latest
+  - name: my-third-public-image
+    image: ubuntu:bionic
+```
+
+## Upgrade
+
+To run the pre-pull again, there are a few options:
+```
+# by default, every upgrade is randomized / does a rollout (`--set randomize=false` to disable this behavior)
+helm upgrade install preepull-daemonset rstudio/prepull-daemonset
+
+# or use kubectl
+kubectl rollout restart daemonset/prepull-daemonset
+```
```

Comment:
What are the steps a user would need to follow when an new image becomes available? Should they run `helm upgrade` or `kubectl rollout` on a cron to fetch the latest on some interval or is there a better way to make sure the cache stays warm?

Reply:
One option for this might be to use a `sleep` for a configurable interval in the `pause` container. Since the daemonset defaults to `restartPolicy: Always` then it should just restart and pull the images again

Reply:
That's a great call. To date we have ignored that question. If you `helm upgrade` (even with same values) the `randomize: true` setting will automatically force a re-pull. However, a way to "keep containers pulled" could be useful.

One downside of doing that is that kubernetes tries to ensure that disks do not get overwhelmed. If you put too many images into the pull or if lots of other images are in use, you could end up overwhelming your disks and cause a bunch of flapping/turnover in the image cache. I'll create an issue to follow-up!

Reply:
Whichever direction we decide to go, I think just an short note in the readme about how/when to run the rollout would be useful. It wasn't immediately clear to me when/ why I would need to run `helm upgrade`